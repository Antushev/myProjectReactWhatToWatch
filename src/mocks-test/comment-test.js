const comments = [
  {
    id: 1,
    user: `Джон`,
    rating: 7.4,
    comment: `Хороший фильм на один раз`,
    date: `July 28, 2020`
  },
  {
    id: 2,
    user: `Андрей`,
    rating: 9,
    comment: `Хороший фильм на четыре раза`,
    date: `July 28, 2020`
  },
  {
    id: 3,
    user: `Джон`,
    rating: 8.8,
    comment: `Хороший фильм на один раз`,
    date: `July 28, 2020`
  },
  {
    id: 4,
    user: `Джон`,
    rating: 1.5,
    comment: `Хороший фильм на один раз`,
    date: `July 28, 2020`
  }
];

export {comments};
