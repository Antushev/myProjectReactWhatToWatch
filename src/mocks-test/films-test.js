import {generateFilm, generateFilms} from '../mocks/films.js';

const FILMS_NUMBER = 10;

const film = generateFilm(1);
const films = generateFilms(FILMS_NUMBER);

export {film, films};
