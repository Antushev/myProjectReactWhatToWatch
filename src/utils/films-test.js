const films = [
  {
    id: 1,
    name: `Хоббит: Пустошь Смауга`,
    picture: `img/pulp-fiction.jpg`,
    genre: `Adventure`,
    date: 2001
  },
  {
    id: 2,
    name: `Интерстеллар`,
    picture: `img/pulp-fiction.jpg`,
    genre: `Adventure`,
    date: 2000
  },
  {
    id: 3,
    name: `Спасти рядоого Райана`,
    picture: `img/pulp-fiction.jpg`,
    genre: `Adventure`,
    date: 2004
  },
  {
    id: 4,
    name: `Хоббит: Пустошь Смауга`,
    picture: `img/pulp-fiction.jpg`,
    genre: `Adventure`,
    date: 2014
  },
  {
    id: 5,
    name: `Власталин колец: Возвращение короля`,
    picture: `img/pulp-fiction.jpg`,
    genre: `Adventure`,
    date: 2014
  },
  {
    id: 6,
    name: `Хоббит: Пустошь Смауга`,
    picture: `img/pulp-fiction.jpg`,
    genre: `Adventure`,
    date: 2019
  },
  {
    id: 7,
    name: `Хоббит: Нежданное путешествие`,
    picture: `img/pulp-fiction.jpg`,
    genre: `Adventure`,
    date: 2020
  },
  {
    id: 8,
    name: `Хоббит: Пустошь Смауга`,
    picture: `img/pulp-fiction.jpg`,
    genre: `Adventure`,
    date: 2018
  }
];

const film = {
  id: 1,
  name: `Интерстеллар`,
  picture: `img/pictrue.jpg`,
  genre: `Adventure`,
  date: 2014
};

export {film, films};
