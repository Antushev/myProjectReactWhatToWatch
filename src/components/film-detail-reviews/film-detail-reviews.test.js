import React from 'react';
import renderer from 'react-test-renderer';

import {film} from '../../mocks-test/films-test.js';
import {comments} from '../../mocks-test/comment-test.js';

import FilmDetailReviews from './film-detail-reviews.jsx';

describe(`FilmDetailReviewsComponent`, () => {
  it(`FilmDetailReviewsComponentSnapshot`, () => {
    const tree = renderer
      .create(
          <FilmDetailReviews
            film={film}
            comments={comments}
            renderTabs={() => {}}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
