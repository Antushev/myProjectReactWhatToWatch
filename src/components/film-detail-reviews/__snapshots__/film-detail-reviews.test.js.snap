// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmDetailReviewsComponent FilmDetailReviewsComponentSnapshot 1`] = `
<div
  className="movie-card__wrap movie-card__translate-top"
>
  <div
    className="movie-card__info"
  >
    <div
      className="movie-card__poster movie-card__poster--big"
    >
      <img
        alt="The Grand Budapest Hotel poster"
        height="327"
        src="img/aviator.jpg"
        width="218"
      />
    </div>
    <div
      className="movie-card__desc"
    >
      <div
        className="movie-card__reviews movie-card__row"
      >
        <div
          className="movie-card__reviews-col"
        >
          <div
            className="review"
          >
            <blockquote
              className="review__quote"
            >
              <p
                className="review__text"
              >
                Хороший фильм на один раз
              </p>
              <footer
                className="review__details"
              >
                <cite
                  className="review__author"
                >
                  Джон
                </cite>
                <time
                  className="review__date"
                  dateTime="{date}"
                >
                  July 28, 2020
                </time>
              </footer>
            </blockquote>
            <div
              className="review__rating"
            >
              7.4
            </div>
          </div>
          <div
            className="review"
          >
            <blockquote
              className="review__quote"
            >
              <p
                className="review__text"
              >
                Хороший фильм на четыре раза
              </p>
              <footer
                className="review__details"
              >
                <cite
                  className="review__author"
                >
                  Андрей
                </cite>
                <time
                  className="review__date"
                  dateTime="{date}"
                >
                  July 28, 2020
                </time>
              </footer>
            </blockquote>
            <div
              className="review__rating"
            >
              9
            </div>
          </div>
          <div
            className="review"
          >
            <blockquote
              className="review__quote"
            >
              <p
                className="review__text"
              >
                Хороший фильм на один раз
              </p>
              <footer
                className="review__details"
              >
                <cite
                  className="review__author"
                >
                  Джон
                </cite>
                <time
                  className="review__date"
                  dateTime="{date}"
                >
                  July 28, 2020
                </time>
              </footer>
            </blockquote>
            <div
              className="review__rating"
            >
              8.8
            </div>
          </div>
          <div
            className="review"
          >
            <blockquote
              className="review__quote"
            >
              <p
                className="review__text"
              >
                Хороший фильм на один раз
              </p>
              <footer
                className="review__details"
              >
                <cite
                  className="review__author"
                >
                  Джон
                </cite>
                <time
                  className="review__date"
                  dateTime="{date}"
                >
                  July 28, 2020
                </time>
              </footer>
            </blockquote>
            <div
              className="review__rating"
            >
              1.5
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
